docs/01-Protocol-Versions/Version5.md

```diff
@@ -0,0 +1,206 @@
+# Paseto Version 5
+
+## GetNonce
+
+Throw an exception. We don't do this in version 5.
+
+## Encrypt
+
+Given a message `m`, key `k`, and optional footer `f` (which defaults to empty
+string), and an optional implicit assertion `i` (which defaults to empty string):
+
+1. Before encrypting, first assert that the key being used is intended for use
+   with `v5.local` tokens, and has a length of 256 bits (32 bytes).
+   See [Algorithm Lucidity](../02-Implementation-Guide/03-Algorithm-Lucidity.md)
+   for more information.
+2. Set header `h` to `v5.local.`
+   **Note**: This includes the trailing period.
+3. Generate 32 random bytes from the OS's CSPRNG to get the nonce, `n`.
+4. Split the key into an Encryption key (`Ek`) and Authentication key (`Ak`),
+   using HKDF-HMAC-SHA384, with `n` appended to the info rather than the salt.
+    * The output length **MUST** be 48 for both key derivations.
+    * The derived key will be the leftmost 32 bytes of the first HKDF derivation.
+
+   The remaining 16 bytes of the first key derivation (from which `Ek` is derived)
+   will be used as a counter nonce (`n2`):
+   ```
+   tmp = hkdf_sha384(
+       len = 48,
+       ikm = k,
+       info = "paseto-encryption-key" || n,
+       salt = NULL
+   );
+   Ek = tmp[0:32]
+   n2 = tmp[32:]
+   Ak = hkdf_sha384(
+       len = 48,
+       ikm = k,
+       info = "paseto-auth-key-for-aead" || n,
+       salt = NULL
+   );
+   ```
+5. Encrypt the message using `AES-256-CTR`, using `Ek` as the key and `n2` as the nonce.
```

Comment:
Could we perhaps clarify what endianness `n2` is meant to be interpreted as for use with AES-256-CTR?

Reply:
Whatever PHP's ext/openssl does with it.

Reply:
See #37